### ISSUE TITLE
using protected methods in Caffe

### ISSUE THREAD

**Issue by robertnishihara** [bytedeco/javacpp-presets#143]
Hi, I'd like to do something like the following:

```scala
import org.bytedeco.javacpp.caffe._

class CaffeSolver(solverParam: SolverParameter) extends FloatSGDSolver(solverParam) {
    super.ApplyUpdate()
}
```

`ApplyUpdate` is a protected method in Caffe's `SGDSolver` class [(link to Caffe code)](https://github.com/BVLC/caffe/blob/16de3407a512ee0eaab8b6c8a60723a4b8a2a8d8/include/caffe/sgd_solvers.hpp). However, `ApplyUpdate` and the other protected methods do not appear in the `FloatSGDSolver` class in [caffe.java](https://github.com/robertnishihara/javacpp-presets/blob/master/caffe/src/main/java/org/bytedeco/javacpp/caffe.java). Would it be possible to wrap those methods as well?

As a short-term fix, I tried the following:

(1) Edit `caffe.java` by inserting the line
```java
@Virtual protected native void ApplyUpdate();
```
in the `FloatSGDSolver` class.
(2) Set `generate-sources.skip` to `true` in `javacpp-presets/pom.xml`.
(3) Run `mvn install --projects .,opencv,caffe`.

However in step (3), I ran into a bunch of errors along the lines of

```
/root/javacpp-presets/caffe/target/classes/org/bytedeco/javacpp/jnicaffe.cpp:1889:5: error: ‘JavaCPP__0003a_0003acaffe_0003a_0003aSGDSolver_0003cdouble_0003e::JavaCPP__0003a_0003acaffe_0003a_0003aSGDSolver_0003cdouble_0003e(std::string&)’ cannot be overloaded
     JavaCPP__0003a_0003acaffe_0003a_0003aSGDSolver_0003cdouble_0003e(std::string& arg0) : ::caffe::SGDSolver<double>(arg0), obj(NULL) { }
     ^
```

Any help/advice would be appreciated. Thanks!

**Comment by saudet** [comment]
Should be fixed in with the commits above! Thanks for reporting!

**Comment by robertnishihara** [comment]
After rebuilding javacpp and javacpp-presets, this seems to work for me. Thanks a lot! Do you have any estimate for when this will be included in the binary release?

**Comment by saudet** [comment]
Next release? Probably around April.... It would go a lot easier if we had a CI server for that though... We could also upload SNAPSHOT binaries for now, good enough?

**Comment by robertnishihara** [comment]
Do you mean upload them to the central maven repository? If so, that'd certainly work.

**Comment by saudet** [comment]
Well here https://oss.sonatype.org/content/repositories/snapshots/org/bytedeco/?

**Comment by robertnishihara** [comment]
Thanks! I'll actually put them up myself in a temporary location, which should be good enough for now.

**Comment by saudet** [comment]
Fix included in newly released version 1.2. Thanks for reporting!
